crates/ty_project/resources/test/corpus/ty_extensions.py

```diff
@@ -0,0 +1,30 @@
+"""
+Make sure that types are inferred for all subexpressions of the following
+annotations involving ty_extension `_SpecialForm`s.
+
+This is a regression test for https://github.com/astral-sh/ty/issues/366
+"""
+
+from ty_extensions import CallableTypeOf, Intersection, Not, TypeOf
+
+
+class A: ...
+
+
+class B: ...
+
+
+def _(x: Not[A]):
+    pass
+
+
+def _(x: Intersection[A], y: Intersection[A, B]):
```

Comment:
The single-argument variant needs to be checked separately, as it takes a different code path. It's important *not* to store types for the single-argument variant, as that happens implicitly in the `self.infer_type_expression(element)` call a few lines above.